misc/language-tour-guide/src/utils/hooks/useDefaultCode.ts

```diff
@@ -0,0 +1,28 @@
+import { useEffect } from "react";
+import { useCodeEditorStore } from "../../components/store/codeEditor";
+import { useLocation } from "react-router";
+import { getNode } from "../content/tree-builder";
+
+export function useDefaultCode() {
```

Comment:
I feel like this is a wrong abstraction, it hides a lot of global state management inside a hook.